clang/test/Driver/arch-arm64e-abi-versioning.c

```diff
@@ -0,0 +1,16 @@
+// Check the ABI version support.
+
+// RUN: %clang -fptrauth-abi-version=5                              -arch arm64e -c %s -### 2>&1 | FileCheck %s --check-prefix ABIVERSION --check-prefix NOKERNELABIVERSION
```

Comment:
@ahmedbougacha This test and the defaults version above are failing on Linux:

```
warning: argument unused during compilation: '-fptrauth-abi-version=5' [-Wunused-command-line-argument]
warning: argument unused during compilation: '-arch arm64e' [-Wunused-command-line-argument]
```

 The `-arch` flag seems to be Darwin specific: https://github.com/apple/llvm-project/blob/next/clang/docs/CommandGuide/clang.rst, and at a glance the new ptrauth ABI related flags are also only used on Darwin. 

Can we gate these tests behind a `REQUIRES: system-darwin` or otherwise fix them?

Reply:
Yep!  Removed -arch in favor of -target in:
  https://github.com/apple/llvm-project/pull/7071
  https://github.com/apple/llvm-project/pull/7072